### PR TITLE
Report various rust-xcb issues to RustSec

### PR DESCRIPTION
**Pull request by psychon**
Closes: https://github.com/RustSec/advisory-db/issues/653

---

As requested by @Shnatsel, here is my attempt at an advisory. I tried to follow what #575 does, but elaborate all the various issues in sub-headings. I did not include `informational = "unsound"` (#575 has this), because this is also `informational = "unmaintained"` and this `informational` entry seems to be undocumented (I only found its use in various advisories), so I am not sure what to do. My best idea: If wanted/necessary, I can split this into two PRs, one for the maintenance status and one for the other issues.

---

I'm also nervous about "there is a code generator that turns XML into lots and lots of `unsafe` code, most of that in functions that are supposed to be safe", but that's likely not worth an advisory. At least this makes it a lot harder to review the code in this crate since I first have to find the actual code in `target/`. Just saying.

(I suppose `cargo geiger` would warn users about this, at least)

---

I take this chance to say "Thank you!" to @Shnatsel. I saw your nick name already a couple of times and my first association to "Shnatsel" is "lots of work for Rust security", like fuzzing, advisory, code review infrastructure etc. If someone else had pinged me and asked for a PR, I am not sure what I would have done, but this is my chance to help you helping Rust. Thanks for all your work.

### CODE REVIEW

**Comment by Shnatsel** [comment]
**TL;DR:** I'm going to drop the "unmaintained" part from this advisory and merge it.

I believe it would be best to make a separate advisory for the maintenance status. That's where `informational = "unmaintained"` would be appropriate. See https://rustsec.org/advisories/RUSTSEC-2016-0005.html or https://rustsec.org/advisories/RUSTSEC-2017-0007.html for examples of how those are usually handled.

Thanks for pointing out the lack of docs, I've opened #751 about this!

And thank you for the kind words! In case you're wondering, here's a sneak peek at what I'm currently working on:

 * https://github.com/rust-secure-code/cargo-supply-chain
 * https://github.com/algesten/ureq/issues/312

**Comment by Shnatsel** [comment]
Done. Thanks!
